nima/tests/integration/webserver/webserver/src/test/java/io/helidon/nima/tests/integration/server/ErrorHandlingWithOutputStreamTest.java

```diff
@@ -0,0 +1,145 @@
+ */
+
+package io.helidon.nima.tests.integration.server;
+
+import io.helidon.common.buffers.DataReader;
+import io.helidon.common.http.Headers;
+import io.helidon.common.http.Http;
+import io.helidon.nima.testing.junit5.webserver.ServerTest;
+import io.helidon.nima.testing.junit5.webserver.SetUpRoute;
+import io.helidon.nima.webclient.http1.Http1Client;
+import io.helidon.nima.webclient.http1.Http1ClientResponse;
+import io.helidon.nima.webserver.http.ErrorHandler;
+import io.helidon.nima.webserver.http.HttpRouting;
+import io.helidon.nima.webserver.http.ServerRequest;
+import io.helidon.nima.webserver.http.ServerResponse;
+import org.junit.jupiter.api.Test;
+
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ServerTest
+class ErrorHandlingWithOutputStreamTest {
+
+    private final Http1Client client;
+
+    ErrorHandlingWithOutputStreamTest(Http1Client client) {
+        this.client = client;
+    }
+
+    @SetUpRoute
+    static void router(HttpRouting.Builder router) {
+        router.error(CustomException.class, new CustomRoutingHandler())
+                .get("get-outputStream", (req, res) -> {
+                    res.outputStream();
+                    throw new CustomException();
+                })
+                .get("get-outputStream-writeOnceThenError", (req, res) -> {
+                    OutputStream os = res.outputStream();
+                    os.write("writeOnceOnly".getBytes(StandardCharsets.UTF_8));
+                    throw new CustomException();
+                })
+                .get("get-outputStream-writeTwiceThenError", (req, res) -> {
+                    OutputStream os = res.outputStream();
+                    os.write("writeOnce".getBytes(StandardCharsets.UTF_8));
+                    os.write("|writeTwice".getBytes(StandardCharsets.UTF_8));
+                    throw new CustomException();
+                })
+                .get("get-outputStream-writeFlushThenError", (req, res) -> {
+                    OutputStream os = res.outputStream();
+                    os.write("writeOnce".getBytes(StandardCharsets.UTF_8));
+                    os.flush();
+                    throw new CustomException();
+                })
+                .get((req, res) -> res.send("ok"));
+    }
+
+    @Test
+    void testOk() {
+        var response = client.get().request();
+
+        assertThat(response.status(), is(Http.Status.OK_200));
+        assertThat(response.entity().as(String.class), is("ok"));
+    }
+
+    @Test
+    void testGetOutputStreamThenError_expect_CustomErrorHandlerMessage() {
+        var response = client.get("/get-outputStream").request();
+
+        assertThat(response.status(), is(Http.Status.I_AM_A_TEAPOT_418));
+        assertThat(response.entity().as(String.class), is("CustomErrorContent"));
+    }
```

Comment:
I should improve the tests by asserting that we should see headers from the "main" handler and only headers from the error handler